### PR TITLE
Support for retrieving user based on client

### PR DESCRIPTION
**Pull request by jorenvandeweyer**
## Summary

Pass the client to the `getUser` function of the model. This makes it possible to retrieve the user based on the client.

## Linked issue(s)

#114 

## Involved parts of the project

Password Grant Type

## Added tests?

Yes

### CODE REVIEW

File: docs/model/spec.rst [hunk 1]
```diff
@@ -454,13 +454,15 @@ This model function is **required** if the ``password`` grant is used.
...
-| password   | String   | The user's password.                                                |
-+------------+----------+---------------------------------------------------------------------+
++-------------------+----------+---------------------------------------------------------------------+
+| Name              | Type     | Description                                                         |
++===================+==========+=====================================================================+
+| username          | String   | The username of the user to retrieve.                               |
++-------------------+----------+---------------------------------------------------------------------+
+| password          | String   | The user's password.                                                |
++-------------------+----------+---------------------------------------------------------------------+
+| client (optional) | Client   | The user's password.                                                |
```
Comment (jankapunkt) [inline docs/model/spec.rst:464]: The description still says "The user's password." Also I think in this docs we don't have "Client" defined as in TypeScript, it should rather be an object, right?

**Review by jankapunkt — CHANGES_REQUESTED** [review]
Generally approved. Please update the spec file and then we can merge.

File: docs/model/spec.rst (hunk 1 above)
Comment (jorenvandeweyer) [reply docs/model/spec.rst]: Client is already defined in some other functions like `saveToken`.

Comment (jorenvandeweyer) [reply docs/model/spec.rst]: Fixed the typo

**Review by jankapunkt — APPROVED** [review]

**Pull request merged**